Fix flake8 errors on 1.0.latest

Since 7/30 there have been flake8 errors.

```
core/dbt/clients/agate_helper.py:20:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
core/dbt/contracts/graph/manifest.py:483:13: F811 redefinition of unused 'filter' from line 481
core/dbt/events/functions.py:195:8: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
core/dbt/events/functions.py:215:8: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
core/dbt/events/types.py:577:16: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
core/dbt/parser/schemas.py:235:16: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
core/dbt/parser/schemas.py:237:[18](https://github.com/dbt-labs/dbt-core/actions/runs/5705325871/job/15459877357#step:5:19): E7[21](https://github.com/dbt-labs/dbt-core/actions/runs/5705325871/job/15459877357#step:5:22) do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
flake8: exit 1 (3.66 seconds) /home/runner/work/dbt-core/dbt-core> flake8 --select=E,W,F --ignore=W504,
```

example run: https://github.com/dbt-labs/dbt-core/actions/runs/5705325871

Since 1.0 is EOL, ignore these errors instead of fixing them.  This allow our scheduled test to get back into a passing state so we can catch actual errors.

--- a/core/dbt/clients/agate_helper.py
+++ b/core/dbt/clients/agate_helper.py
@@ -17,7 +17,7 @@ class Number(agate.data_types.Number):
     # undo the change in https://github.com/wireservice/agate/pull/733
     # i.e. do not cast True and False to numeric 1 and 0
     def cast(self, d):
-        if type(d) == bool:
+        if type(d) == bool:  # noqa: E721
             raise agate.exceptions.CastError(
                 'Do not cast True to 1 or False to 0.'
             )

--- a/core/dbt/contracts/graph/manifest.py
+++ b/core/dbt/contracts/graph/manifest.py
@@ -480,7 +480,7 @@ class MacroMethods:
         """
         filter: Optional[Callable[[MacroCandidate], bool]] = None
         if package is not None:
-            def filter(candidate: MacroCandidate) -> bool:
+            def filter(candidate: MacroCandidate) -> bool:  # noqa: F811
                 return package == candidate.macro.package_name
 
         candidates: CandidateList = self._find_macros_by_name(

--- a/core/dbt/events/functions.py
+++ b/core/dbt/events/functions.py
@@ -192,7 +192,7 @@ def create_info_text_log_line(e: T_Event, msg_fn: Callable[[T_Event], str]) -> s
 def create_debug_text_log_line(e: T_Event, msg_fn: Callable[[T_Event], str]) -> str:
     log_line: str = ''
     # Create a separator if this is the beginning of an invocation
-    if type(e) == MainReportVersion:
+    if type(e) == MainReportVersion:  # noqa: E721
         separator = 30 * '='
         log_line = f'\n\n{separator} {e.get_ts()} | {get_invocation_id()} {separator}\n'
     color_tag: str = '' if this.format_color else Style.RESET_ALL
@@ -212,7 +212,7 @@ def create_debug_text_log_line(e: T_Event, msg_fn: Callable[[T_Event], str]) -> 
 # translates an Event to a completely formatted json log line
 # you have to specify which message you want. (i.e. - e.message(), e.cli_msg(), e.file_msg())
 def create_json_log_line(e: T_Event, msg_fn: Callable[[T_Event], str]) -> Optional[str]:
-    if type(e) == EmptyLine:
+    if type(e) == EmptyLine:  # noqa: E721
         return None  # will not be sent to logger
     # using preformatted string instead of formatting it here to be extra careful about timezone
     values = event_to_serializable_dict(e, lambda _: e.get_ts_rfc3339(), lambda x: msg_fn(x))

--- a/core/dbt/events/types.py
+++ b/core/dbt/events/types.py
@@ -574,7 +574,7 @@ class ListRelations(DebugLevel, Cli, File):
     def asdict(cls, data: list) -> dict:
         d = dict()
         for k, v in data:
-            if type(v) == list:
+            if type(v) == list:  # noqa: E721
                 d[k] = [str(x) for x in v]
         return d
 

--- a/core/dbt/parser/schemas.py
+++ b/core/dbt/parser/schemas.py
@@ -232,9 +232,9 @@ class SchemaParser(SimpleParser[GenericTestBlock, ParsedGenericTestNode]):
         def get_hashable_md(
             data: Union[str, int, float, List, Dict]
         ) -> Union[str, List, Dict]:
-            if type(data) == dict:
+            if type(data) == dict:  # noqa: E721
                 return {k: get_hashable_md(data[k]) for k in sorted(data.keys())}  # type: ignore
-            elif type(data) == list:
+            elif type(data) == list:  # noqa: E721
                 return [get_hashable_md(val) for val in data]  # type: ignore
             else:
                 return str(data)

